Cesium.CodeGen/Ir/Expressions/FunctionCallExpression.cs

```diff
@@ -32,9 +33,46 @@ public FunctionCallExpression(Ast.FunctionCallExpression expression)
         _arguments = (IList<IExpression>?)arguments?.Select(e => e.ToIntermediate()).ToList()
                      ?? Array.Empty<IExpression>();
         _callee = null;
+        _castLike = castLike;
     }
 
     public IExpression Lower(IDeclarationScope scope)
+    {
+        if (_castLike)
```

Comment:
Is is possible that `castLike` is not needed after #365 ? The only concern is `size_t c = (foo)(1, 2);` but `(foo)` is function address expression, so I do not see that it need special handling like `castLike`

Reply:
I look at at closer and understand the problem. I think better way, or maybe that y proference is to introduce new node 
```
record class FunctionCallOrCastExpression(function, arguments);
```

which in the lower phase will try to perform all checks and detect is it function call or type and create new nodes `FuncntionCallExpression` respectively and `CastExpression` respectively and lower them inside. That way IMO we have clearly function calls and clearly cast expression and some muddy syntax details would be encapsulated in one place.